documentation/docs/developer_guide/coding_standards.md

```diff
@@ -1 +1,160 @@
+
+Submitted developments that are inconsistent with the standards discussed below, and/or poorly commented/documented code will not be accepted without revision. 
+## Coding Standards
+Generic code examples are given below as well as a link to the JULES coding standards, to which CABLE is somewhat bound through the JULES and CABLE (JAC) project.  
+Whilst syntax/text formatting style is relatively trivial, it must still be adhered to. Some more significant general rules that should be followed are: 
+## The JAC project 
+The JAC project is a long term initiative to include CABLE as an inter-operable Land-Surface Model (LSM) within the Unified Model (UM). For several reasons, CABLE is coupled to the UM via its native LSM, JULES. From a coding standards perspective, both the UM and JULES developments are strictly goverened by the UK Met Office and thus CABLE's coding standards are somewhat constrained by JULES' coding standards. 
+
+The [JULES coding standards](http://jules-lsm.github.io/coding_standards/) are described in this comprehensive document.
+### 1. IO must be restricted to the top level
+In terms of the offline application of the model this means IO should be restricted to the offline/ directory. Coupled to the UM/JULES there should be no IO code in the CABLE directory. 
+
+### 2. Comment liberally
+Detrimental to any model is the lack of meaningful and/or outdated comments. 
+Liberal inline commenting is expected aswell as thorough commenting in the MarkDown. See [cable_roughness.F90](https://github.com/CABLE-LSM/CABLE/blob/main/src/science/roughness/cable_roughness.F90) for an example, which is then redered as ?????
+### 3. Avoid WHERE loops
+WHERE loops effectively replace DO/IF loops. In some circumstances these may improve readability. They certainly make it easier to write code. However they are nightmare to debug and hamper portability. Especially nested WHERE loops. 
+Moreover it will not pass JULES coding standards. 
+### 4. Variable names
+We are not so limited by the number of character spaces available as we once were. Developers are encouraged to name variables meaningfully. Of course commenting might make it clear what the variable corresponds to however it is helpful to be able to understand what is going on whilst reading a line of code.
+
+Please maintain globally consistent names of variables across files/subroutines/modules. 
+### 5. File names
+To include CABLE code identifiably across applications, please include the "_cbl" suffix in your filename **eg. egName_cbl.F90**. 
+
+Please maintain consistent naming between files/subroutines/modules. 
+### 6. MODULE names
+Everything should be contained within modules.
+Modules should be of the form "*_mod_cbl". Following from the above file egName_cbl.F90 **MODULE egName_mod_cbl**.
+
+### 7. SUBROUTINE names
+Subroutines should be of the form "*_cbl()". Following from the above file egName_cbl.F90 **SUBROUTINE egName_cbl()**.
```

Comment:
Honestly, my memory of how adamant JULES have been about this is wavering. They have flip-flopped over the years and instances of all permutations can be found. It is certainly true that it is very helpful to be able to search for _cbl and/or identify a CABLE file simply from its name. How far this applies to subroutines is more dubious.  It can be helpful, for e.g. in 10,000 line job.err files which might identify the subroutine where it crashed. There are several instances (and more to come) where we have a CABLE-alagous version of a JULES subroutine. An example,
``` 
IF (lsm_id == JULES) THEN
   CALL screen_temp() 
ELSE  
   CALL screen_temp_cbl() 
END IF
```
@ccarouge was the strongest proponent for file naming. I don't really care, but there is no doubt that it forces consistency and there isn't really a downside.

Reply:
Okay, I can see now how it would be useful. Can you mention here something along those lines? Like "... which distinguishes CABLE versions of subroutines that are analogous to JULES subroutines."